app/demo-loop/demo-loop.cc

```diff
@@ -83,7 +84,10 @@ void run(std::istream* is, std::shared_ptr<celeritas::OutputRegistry> output)
     double const setup_time = get_setup_time();
 
     // Run on a single thread
-    RunnerResult result = run_stream(celeritas::StreamId{0});
+    RunnerResult result = [&run_stream] {
+        celeritas::ScopedDeviceProfiling profile_this;
+        return run_stream(celeritas::StreamId{0});
+    }();
```

Comment:
So this effectively profile the entire app, what nsys and ncu do by default?

Reply:
No, it starts the profiling after construction. VecGeom literally launches one kernel per geometry volume and that shit gets captured by default.

Reply:
Just FYI: there are some VecGeom commits from June 2021, which imply a factor 5x speed up in geometry transfer to the device, by transferring all geometry parameters to the GPU as a block, and then launching PlacedVolume construction kernels in parallel for each shape type.  See VecGeom commits 5d8be4866 and 4c0d19fb0.

Reply:
Interesting... is there something different we have to do to activate it? I was seeing an interminable number of kernels when setting up the cms2018 geometry.

Reply:
It should be active, and I'd be surprised if the old version would stay active once there is a faster version.
According to the code comments, there are still multiple kernels, basically one per basic shape type, which should construct all needed instances of that shape in parallel, and for some complex shapes which still require one kernel call per shape instance.  I don't know details though.